Initial Implementation

rsandor opened the pull request:
Attempt to modularize our copy-pasta `error.js` files. Inspired by Navi's `error.js`.

Commit:
Initial implementation.

Commit:
Fleshed out the readme.

podviaznikov commented on index.js:
```diff
@@ -0,0 +1,113 @@
+ * @param {Error} err The error to report.
+ */
+ErrorCat.prototype.report = function (err) {
+  if (this.canUseRollbar()) {
```
maybe also add `report` flag. https://github.com/CodeNow/api/blob/master/lib/error.js#L90

podviaznikov commented on index.js:
```diff
@@ -0,0 +1,113 @@
+ */
+ErrorCat.prototype.report = function (err) {
+  if (this.canUseRollbar()) {
+    rollbar.handleErrorWithPayloadData(err, { custom: (err.data || {}) }, noop);
```
`err.data` willl fail if `err` undefined

podviaznikov commented on index.js:
```diff
@@ -0,0 +1,113 @@
+ * Express error responder middleware.
+ * @type {function}
+ */
+Object.defineProperty(ErrorCat, 'responder', {
```
Don't really like this name. Too general. Maybe call  it `error-middleware` or something in that direction? Just thoughts

rsandor replied:
What about just `middleware`?

rsandor replied:
Then it would be

``` js
app.use(ErrorCat.middleware);
```

podviaznikov replied:
works for me

podviaznikov commented on package.json:
```diff
@@ -0,0 +1,45 @@
+    "test": "test"
+  },
+  "scripts": {
+    "test": "jshint index.js && lab -v -c -a code test/*"
```
maybe create separate commands: `unit` and `lint` and then piper them together?
@bkendall would appreciate

rsandor replied:
Can you give me an example of a project that does this?

podviaznikov replied:
https://github.com/CodeNow/api/blob/master/package.json#L133

Commit:
Now checking error report flag.

Commit:
Renamed `responder` to `middleware`

- Removed `ErrorCat._instance` property.
- Better, more modular, build scripts.

Commit:
Added jsdoc script and contributing section to README.

Commit:
Added TravisCI.

podviaznikov commented:
looks good to me

Commit:
Added TravisCI badge.

bkendall commented:
tests!!! <3

Commit:
Added dependency badges.

bkendall commented:
:+1: LGTM
